src/use-visitor-data.ts

```diff
@@ -31,13 +30,14 @@ export function useVisitorData<TExtended extends boolean>(
 ): VisitorQueryContext<TExtended> {
   assertIsTruthy(getOptions, 'getOptions')
 
-  const previousGetOptions = usePrevious(getOptions)
-
   const { immediate } = config
   const { getVisitorData } = useContext<FpjsContextInterface<TExtended>>(FpjsContext)
 
-  const initialState = { isLoading: config.immediate ? true : false }
-  const [state, setState] = useState<QueryResult<VisitorData<TExtended>>>(initialState)
+  const initialState: QueryResult<VisitorData<TExtended>> & { getOptions: UseVisitorDataOptions<TExtended> } = {
+    isLoading: config.immediate ? true : false,
```

Comment:
@necipallef suggested that this could be simplified to:
```suggestion
    isLoading: config.immediate,
```